### ISSUE TITLE
consider pulling out sequential integration into a testable function

### ISSUE THREAD

**Issue by klaeufer** [LoyolaChicagoCode/unoapi-dpcpp-examples#10]

**Comment by gkthiruvathukal** [comment]
@klaeufer I like this idea.

One other thing I am interested to know is whether a unit test can be performed "on device" for a "parallel_for). This won't necessarily work (well) in the CI pipeline, but if we can get it to work when doing standalone work, say, on DevCloud, this would be awesome. Given that host/device arch can be different in general, we want tests to pass in both worlds.

I believe it can work. We just need to make sure the text fixture can do all needed setup for the parallel part. i realize this should probably be a separate issue (GitHub wise).

**Comment by klaeufer** [comment]
We could unit-test the entire inner lambda, but this would require pulling it out into a suitable template function (similar to print_function_values).

```
q.submit([&](auto & h) {
const sycl::accessor v{v_buf, h};
h.parallel_for(size, [=](const auto & index) {
v[index] = f(x_min + index * dx);
}); // end inner lambda/parallel_for
}); // end of command group
```

Alas, we wouldn't be able to test the reduce stage because of the SYCL dependency.

```
h.parallel_for(
sycl::range<1>{number_of_trapezoids},
sum_reduction,
[=](const auto & index, auto & sum) {
sum.combine(
trapezoid(v[index], v[index + 1], half_dx));
});
```

**Comment by klaeufer** [comment]
We'll soon do this for the new `compute_outer_trapezoid` function!

**Comment by klaeufer** [comment]
resolved by commit d8a29cb86c6d9d5a1b849ca137ed5c30a8bf0723
